doc/guides/beyond-GSG.rst

```diff
@@ -0,0 +1,277 @@
+.. _beyond-gsg:
+
+Beyond the Getting Started Guide
+################################
+
+The :ref:`getting_started` gives a straight-forward path to set up
+your Linux, macOS, or Windows enviromnent for Zephyr development. In
+this document, we delve deeper into Zephyr development setup
+issues and alternatives.
+
+.. _python-pip:
+
+Python and pip
+**************
+
+Python 3 and its package manager, pip\ [#pip]_, are used extensively by Zephyr
+to install and run scripts required to compile and run Zephyr
+applications, set up and maintain the Zephyr development environment,
+and build project documentation.
+
+Depending on your operating system, you may need to provide the
+``--user`` flag to the ``pip3`` command when installing new packages. This is
```

Comment:
We could be clearer here and explicitly state that you need the --user for Linux.

Reply:
No because it depends which "Linux".

Reply:
.. for all supported operating systems except for macOS and Windows? ;)
At least that's how it's currently documented.

Reply:
From: https://docs.brew.sh/Homebrew-and-Python#setuptools-pip-etc
> The normal pip install --user is disabled for brewed Python. This is because of a bug in distutils

So this could change on macOS any time after  that bug is fixed.

On Debian and most other debian-based distros (including Ubuntu?):
https://github.com/pypa/pip/issues/4222#issuecomment-417646535
> When running as a normal user in a non-virtual environment, default to
--user and --ignore-installed.  When inside virtual environments or when
running as root, keep the default behavior.

Not the case on other Linux distributions.

It's a mess so there's no way to clarify the "Depending on your operating system" part, you can stop trying. The only solution is to:
1. focus on and try to stay up to date with a *limited* number of operating systems: Ubuntu, macOS, Windows, Period.
2. defer to OS-specific documentations for other (Linux) operating systems